dds/DCPS/Service_Participant.h

```diff
@@ -80,6 +87,8 @@ class OpenDDS_Dcps_Export Service_Participant {
 
   ACE_thread_t reactor_owner() const;
 
+  void add_shutdown_listener(ShutdownListener* listener);
```

Comment:
```suggestion
  void set_shutdown_listener(ShutdownListener* listener);
```
Use "add" if there can be more than one.  Doesn't look like that's the case here.